enhancements/subscription-content/subscription-content-access.md

```diff
@@ -0,0 +1,242 @@
+So that I download RHEL 8 content in UBI 8 containers And download RHEL 7 content in UBI 7
+containers
+
+### Implementation Details/Notes/Constraints [optional]
+
+To consume RHEL content in OpenShift, the running container must have access to the following:
+
+One or more entitlement keys, which are PEM-encoded certificates. A yum repository definition, which
+tells yum/dnf where to access content from.
+
+Yum repository definitions are specific to the major version of RHEL that the running container is
+based on - a RHEL 8-based container is not compatible with RHEL 7 content.
+
+This enhancement proposal will detail at a high level the functions of:
+
+A cluster-wide subscription manager to generate subscription content access data A projected
+resource CSI driver to share resources across OpenShift namespaces A subscription content webhook to
+generate the necessary volume mounts to add subscription content access data to containers
+Extensions to propagate subscription content access data into OpenShift builds.
+
+#### Cluster Subscription Manager
+
+The cluster subscription manager will take as input the `cloud.redhat.com` pull secret, which
+already provides authentication for the cluster and provides the means to register clusters with Red
+Hat’s cloud manager. The pull secret is then used to obtain the following: The `redhat.repo` yum
+repo configurations for all supported major versions of RHEL. Today these will be RHEL 7 and 8 -
+later major versions of RHEL must be anticipated. The Simple Content Access certificate for the
+customer, which will serve as the entitlement key.
+
+As RHSM does today on RHEL hosts, the cluster subscription manager will be responsible for
+generating `redhat.repo` definitions for the currently supported RHEL major versions and saving it
+on the cluster. Each `redhat.repo` file will be saved in a ConfigMap within the `openshift-config`
+namespace, named `redhat-repo-rhel${n}`, where `${n}` is the major version of RHEL associated with
+the `redhat.repo` file.
+
+The cluster subscription manager will also need to obtain the entitlement keys associated with the
+attached OpenShift subscription. It is assumed that OpenShift entitlements grant access to RHEL
+content, but may not grant access to layered content such as EAP or Application Runtimes. This
+content must be saved in a Secret within the `openshift-config` namespace, named
+`etc-pki-entitlement`.
+
+The full details of this component will be specified in a subsequent enhancement proposal.
+
+#### Projected-resource CSI Driver
+
+Once the entitlement key(s) and yum repository definitions are stored on the cluster, the next step
+is to share these definitions across namespaces. This task will fall to the projected-resource
+containers storage interface (CSI) driver. This will be a special storage driver that allows
+ConfigMaps and Secrets to be shared across namespaces via standard pod volume mounts. Access to
```

Comment:
Hi @gabemontero !
I understand what the Projected-resource CSI driver is trying to solve here - It would project the entitlement secret(s) from the `openshift-config` namespace into the namespace where workloads would request for entitlements. 

Is there a reason why we haven't considered a solution where:
1. A controller watches for a specific annotation on a pod/workload in namespace X
2. Creates a copy of the entitlement secret into the namespace X, ownerRef'd to the requesting workload.
3. A mutating webhook mounts that secret as a volume into the workload.

Reply:
and to connect the dots, similar alternatives were discussed and dismissed early enough that it preceded authoring of the EP for the projected resource CSI driver and I now see they didn't even make it to the alternatives section in https://github.com/openshift/enhancements/blob/master/enhancements/cluster-scope-secret-volumes/csi-driver-host-injections.md#alternatives

A follow up PR to list that dismissed alternative in that EP could be done if you feel strongly about it @sbose78

Reply:
Sure, I am happy to make a follow-up PR.

@gabemontero @bparees Would you be able to point me to an existing discussion on this? I'll look for it in the PR/discussion of https://github.com/openshift/enhancements/blob/master/enhancements/cluster-scope-secret-volumes/csi-driver-host-injections.md#alternatives

Reply:
pertty sure it was prior to the PRs going up @sbose78 as I tried to convey with my `...preceded authoring of the EP...` bit in my last comment.... so nothing I can point you to wrt that.

At this point, just add the notion in the alternatives section along with some form of @bparees 's https://github.com/openshift/enhancements/pull/384#discussion_r443730770

we'll then see during PR review of your update if more cons come to mind.

Reply:
@sbose78 @gabemontero Added "Automate Copying" as an alternative.

Reply:
new alternative looks good to me @adambkaplan and addresses the discussion thread here between you, me, @sbose78 , and @bparees IMO

Reply:
Cool!